src/util.rs

```diff
@@ -179,6 +184,55 @@ impl<'r> RepositoryUtil<'r> for Repository {
         trailers
     }
 
+    fn cli_author(&self, matches: &ArgMatches) -> git2::Signature {
+        use chrono::{DateTime, FixedOffset};
+
+        // First, get a signature appropriate for the author
+        let author = matches
+            .value_of("author")
+            .map(|a| {
+                // NOTE: this function will be called only once per run anyway
+                let parts = Regex::new(r"^([^<>]+)[[:space:]]+<([^[:space:]@]+@[^[:space:]@]+)>$")
```

Comment:
This regex could be put either in the outer scope or even as constant into a `lazy_static`, so the regex is only compiled once per call of `cli_author()` or even only once per `git-dit` execution!

Reply:
Have you even read the note one line above? If this function is called more than once per run, the dev using it is doing something wrong.

Reply:
Whops, jup.